File: examples/python/basic.py
"""
Markdown Table Generator API - Basic Usage Example

This example demonstrates the basic usage of the Markdown Table Generator API.
API Documentation: https://docs.apiverve.com/ref/markdowntable
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/markdowntable'

def call_markdowntable_api():
    """
    Make a POST request to the Markdown Table Generator API
    """
    try:
        # Request body
        request_body &#x3D; {
    &#x27;headers&#x27;: [
        &#x27;Name&#x27;,
        &#x27;Age&#x27;,
        &#x27;City&#x27;
    ],
    &#x27;rows&#x27;: [
        [
            &#x27;John Doe&#x27;,
            30,
            &#x27;New York&#x27;
        ],
        [
            &#x27;Jane Smith&#x27;,
            25,
            &#x27;Los Angeles&#x27;
        ],
        [
            &#x27;Bob Johnson&#x27;,
            35,
            &#x27;Chicago&#x27;
        ]
    ],
    &#x27;alignment&#x27;: &#x27;left&#x27;
}

        headers = {
            'x-api-key': API_KEY,
            'Content-Type': 'application/json'
        }

        response = requests.post(API_URL, headers=headers, json=request_body)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Markdown Table Generator API...\n')

    result = call_markdowntable_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: examples/javascript/basic.js
/**
 * Markdown Table Generator API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Markdown Table Generator API.
 * API Documentation: https://docs.apiverve.com/ref/markdowntable
 */

const API_KEY = process.env.APIVERVE_API_KEY || 'YOUR_API_KEY_HERE';
const API_URL = 'https://api.apiverve.com/v1/markdowntable';

/**
 * Make a POST request to the Markdown Table Generator API
 */
async function callMarkdownTableGeneratorAPI() {
  try {
    // Request body
    const requestBody &#x3D; {
    &quot;headers&quot;: [
        &quot;Name&quot;,
        &quot;Age&quot;,
        &quot;City&quot;
    ],
    &quot;rows&quot;: [
        [
            &quot;John Doe&quot;,
            30,
            &quot;New York&quot;
        ],
        [
            &quot;Jane Smith&quot;,
            25,
            &quot;Los Angeles&quot;
        ],
        [
            &quot;Bob Johnson&quot;,
            35,
            &quot;Chicago&quot;
        ]
    ],
    &quot;alignment&quot;: &quot;left&quot;
};

    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'x-api-key': API_KEY,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(requestBody)
    });

    // Check if response is successful
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();

    // Check API response status
    if (data.status === 'ok') {
      console.log('✓ Success!');
      console.log('Response data:', data.data);
      return data.data;
    } else {
      console.error('✗ API Error:', data.error || 'Unknown error');
      return null;
    }

  } catch (error) {
    console.error('✗ Request failed:', error.message);
    return null;
  }
}

// Run the example
callMarkdownTableGeneratorAPI()
  .then(result => {
    if (result) {
      console.log('\n📊 Final Result:');
      console.log(JSON.stringify(result, null, 2));
    }
  });
